components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java

```diff
@@ -2940,41 +2940,52 @@ public static boolean jsFunction_addDocumentation(Context cx, Scriptable thisObj
         FileHostObject fileHostObject = null;
         String sourceURL = null;
 
-        APIIdentifier apiId = new APIIdentifier(APIUtil.replaceEmailDomain(providerName), apiName, version);
-        Documentation doc = new Documentation(getDocType(docType), docName);
-        APIProvider apiProvider = getAPIProvider(thisObj);
+        boolean isTenantFlowStarted = false;
+
+        try {
+            String tenantDomain = MultitenantUtils.getTenantDomain(APIUtil.replaceEmailDomainBack(providerName));
+            if(tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                isTenantFlowStarted = true;
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
+            }
+            APIIdentifier apiId = new APIIdentifier(APIUtil.replaceEmailDomain(providerName), apiName, version);
+            Documentation doc = new Documentation(getDocType(docType), docName);
+            APIProvider apiProvider = getAPIProvider(thisObj);
 
-        //add documentation is allowed only if document name does not already exist for this api
-        if (apiProvider.isDocumentationExist(apiId, docName)) {
-            handleException("Error occurred while adding the document. " + docName +
+            //add documentation is allowed only if document name does not already exist for this api
+            if (apiProvider.isDocumentationExist(apiId, docName)) {
+                handleException("Error occurred while adding the document. " + docName +
                     " already exists for API " + apiName + "-" + version);
-        }
+            }
 
-        if (doc.getType() == DocumentationType.OTHER) {
-            doc.setOtherTypeName(args[9].toString());
-        }
+            if (doc.getType() == DocumentationType.OTHER) {
+                doc.setOtherTypeName(args[9].toString());
+            }
 
-        if (sourceType.equalsIgnoreCase(Documentation.DocumentSourceType.URL.toString())) {
-            doc.setSourceType(Documentation.DocumentSourceType.URL);
-            sourceURL = args[7].toString();
-        } else if (sourceType.equalsIgnoreCase(Documentation.DocumentSourceType.FILE.toString())) {
-            doc.setSourceType(Documentation.DocumentSourceType.FILE);
-            fileHostObject = (FileHostObject) args[8];
-        } else {
-            doc.setSourceType(Documentation.DocumentSourceType.INLINE);
-        }
+            if (Documentation.DocumentSourceType.URL.toString().equalsIgnoreCase(sourceType)) {
```

Comment:
Since the try block defined in line 2945, the code inside that get formatted as this. Hence not an issue.